### PR TITLE
libobs: Add filter hotkeys

### PR DESCRIPTION
**Pull request by cg2121**
This adds hotkeys to enable and disable filters.

### CODE REVIEW

**Comment by Tizye** [comment]
Is this part of a larger coding addition or will it work standalone in OBS Studio? The "files changed" don't seem to be present in my /obs-studio install.

File: libobs/obs-source.c [hunk 1]
```diff
@@ -226,6 +226,34 @@ static bool obs_source_hotkey_unmute(void *data,
...
 
+static bool obs_source_filter_hotkey_enable(void *data,
+		obs_hotkey_pair_id id, obs_hotkey_t *key, bool pressed)
+{
+	UNUSED_PARAMETER(id);
+	UNUSED_PARAMETER(key);
+
+	struct obs_source *source = data;
+
+	if (!pressed || obs_source_enabled(source)) return false;
```
Comment (RytoEX) [inline libobs/obs-source.c]: Style:  Don't put multiple statements on a single line.

File: libobs/obs-source.c [hunk 2]
```diff
@@ -226,6 +226,34 @@ static bool obs_source_hotkey_unmute(void *data,
...
+
+static bool obs_source_filter_hotkey_disable(void *data,
+		obs_hotkey_pair_id id, obs_hotkey_t *key, bool pressed)
+{
+	UNUSED_PARAMETER(id);
+	UNUSED_PARAMETER(key);
+
+	struct obs_source *source = data;
+
+	if (!pressed || !obs_source_enabled(source)) return false;
```
Comment (RytoEX) [inline libobs/obs-source.c]: Style:  Don't put multiple statements on a single line.

File: libobs/obs-source.c [hunk 3]
```diff
@@ -292,6 +320,22 @@ static void obs_source_init_audio_hotkeys(struct obs_source *source)
 			obs_source_hotkey_push_to_talk, source);
 }
 
+static void obs_source_init_filter_hotkeys(struct obs_source *source)
+{
+	if (source->info.type != OBS_SOURCE_TYPE_FILTER) {
+		source->filter_enable_disable_key  = OBS_INVALID_HOTKEY_ID;
```
Comment (RytoEX) [inline libobs/obs-source.c]: There's an extra space before the equal sign.

**Comment by cg2121** [comment]
I've combined this with #1327 because my new PR has much simpler code.

**Pull request closed without merging**